src/sonic-yang-models/yang-models/sonic-vlan-sub-interface.yang

```diff
@@ -0,0 +1,78 @@
+module sonic-vlan-sub-interface {
+
+    yang-version 1.1;
+
+    namespace  "http://github.com/Azure/sonic-vlan-sub-interface";
+
+    prefix vlan-sub-interface;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "VLAN sub interface for SONiC OS";
+
+    revision 2021-11-11 {
+        description "Initial version";
+    }
+
+    container sonic-vlan-sub-interface {
+        container VLAN_SUB_INTERFACE {
+
+            description "VLAN SUB INTERFACE part of config_db.json";
+
+            list VLAN_SUB_INTERFACE_LIST {
+
+                description "VLAN_SUB_INTERFACE part of config_db.json with vrf";
+
+                key "name";
+
+                leaf name {
+                    must "substring-before(current(), '.') = /port:sonic-port/port:PORT/port:PORT_LIST/port:name"
+                    {
+                        error-message "Must condition not satisfied, Try adding PORT: {}, Example: 'Ethernet0': {}";
+                    }
+
+                    type string {
+                        pattern '(\w+)\.([0-9]{1,3}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4])';
```

Comment:
>[0-9]{1,3}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4] [](http://example.com/codeflow?start=41&length=56)

Could you explain the pattern in English in code comment?

Reply:
updated, thanks!

Reply:
@qiluo-msft, could you please give another review?

Reply:
How about `[1-9][0-9]{0,2}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4]` ?

Reply:
> How about `[1-9][0-9]{0,2}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4]` ?

Yes, this will exclude leading zeros, will update, thanks~

Reply:
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-vlan.yang#L148
sonic-vlan.yang is using the same pattern, need to update?